### PR TITLE
feature/save-mix

### PR DESCRIPTION
**Pull request by wolfbiter**
add save mix functionality, related reducers

open question in code TODOs

### CODE REVIEW

File: mixes/reducer.js [hunk 1]
```diff
@@ -62,6 +67,31 @@ function createReducer (config) {
...
+      //       is this an antipattern? should i instead pass mixId to saveMix,
+      //       then in the reducer regenerate the flat mix, then call nestMix?
+      Effects.promise(runSaveMix, action.payload),
+
+      // TODO: if going with this approach, need to somehow get channels, clips
+      // Effects.promise(runSaveMix, nestMix({
+      //   ...action.payload,
+      //   channels: [],
+      //   clips: []
+      // })),
```
Comment (wolfbiter) [inline mixes/reducer.js]: thoughts?

File: mixes/reducer.js [hunk 2]
```diff
@@ -62,6 +67,31 @@ function createReducer (config) {
...
+
+      Effects.constant(saveMixEnd())
+    ])),
+    [saveMixSuccess]: (state, action) => state, // noop, is this OK for that?
```
Comment (wolfbiter) [inline mixes/reducer.js]: thoughts?

File: mixes/service.js [hunk 3]
```diff
@@ -29,4 +30,10 @@ function createService (config) {
       .then(JSON.parse)
       .then(mix => ({ ...mix, id }))
   }
+
+  // TODO: how print formatted json?
```
Comment (wolfbiter) [inline mixes/service.js]: thoughts?

File: mixes/reducer.js (hunk 1 above)
Comment (ahdinosaur) [reply mixes/reducer.js]: your comments are bang on, yet i think for now it's best to keep passing in the pre-nested mix. i kept this (and use cases like this) in mind when [designing my own front-end framework](https://github.com/ahdinosaur/inux/issues/8#issuecomment-267492008), it'd be nice if each reducer was independently scope-able.

File: mixes/reducer.js (hunk 2 above)
Comment (ahdinosaur) [reply mixes/reducer.js]: yeah that's legit.

File: mixes/service.js (hunk 3 above)
Comment (ahdinosaur) [reply mixes/service.js]: to stringify "pretty" JSON (2-space indents):

```js
JSON.stringify(mix, null, 2)
```

[`JSON.stringify` docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify)

**Review by ahdinosaur — APPROVED** [review]
everything is awesome! :watermelon:

**Pull request merged**